app/components/Settings/Preview/TableTree.react.js

```diff
@@ -40,19 +60,44 @@ class TableTree extends Component {
         }
     }
 
-    storeSchemaTree() {
-        const {schemaRequest, getSqlSchema, updatePreview} = this.props;
+    /**
+     * Checks to see whether it is still loading the request for the schema.  If loading
+     * or undefined, returns loading.  If status not 200 returns Error message
+     * @param {object} status - Request Status
+     * @param {object} schemaRequest - Scheme Request
+     * @param {object} treeSchema - Tree Schema
+     * @returns {object} div containing connect message
+     */
+    isConnecting(status, schemaRequest, treeSchema) {
+        if (typeof status === 'undefined' || status === 'loading') {
+            return (<div className="loading">{'Loading'}</div>);
+        }
 
-        if (typeof schemaRequest === 'undefined') {
-            getSqlSchema();
+        if (status !== 200) {
+            return (
+                <div style={{padding: '5px', fontSize: '12px'}}>
+                    {`ERROR ${JSON.stringify(schemaRequest)}`}
+                </div>
+            );
         }
-        else if (isEmpty(schemaRequest)) {
-            getSqlSchema();
+
+        if (!treeSchema) {
+            return (<div className="loading">{'Updating'}</div>);
         }
-        else if (schemaRequest.status === 200 && !has('treeSchema', this.props.preview)) {
-            const treeSchema = {};
+    }
+
+    /**
+     * The following method will create the tree schema
+     * @param {object} schemaRequest - The Schema Request
+     * @returns {object} treeSchema - Formatted Tree Schema
+     */
+    createTreeSchema(schemaRequest) {
+        const treeSchema = {};
+
+        if ((schemaRequest) && (schemaRequest.content) &&
+                (schemaRequest.content.rows) && (schemaRequest.content.rows.length > 0)) {
             schemaRequest.content.rows.forEach(function(row) {
-                const [tableName, columnName, dataType] = row;
+                const {tableName, columnName, dataType} = row;
```

Comment:
I haven't debugged this line, but I'm convinced this change is wrong. `rows` is an array of arrays.